### PR TITLE
expression: implement vectorized evaluation for `builtinYearWeekWithModeSig`

### PR DESCRIPTION
**Pull request by chenlx0**
PCP #12103 

### What problem does this PR solve? 

implement vectorized evaluation for `builtinYearWeekWithModeSig`

### What is changed and how it works?

```
$ make vectorized-bench VB_FILE=Time VB_FUNC=builtinYearWeekWithModeSig
cd ./expression && \
                go test -v -benchmem \
                        -bench=BenchmarkVectorizedBuiltinTimeFunc \
                        -run=BenchmarkVectorizedBuiltinTimeFunc \
                        -args "builtinYearWeekWithModeSig"
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFuncGenerated-8           1000000000               0.00732 ns/op         0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinYearWeekWithModeSig-VecBuiltinFunc-8             28470             41544 ns/op           29192 B/op          8 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinYearWeekWithModeSig-NonVecBuiltinFunc-8          19538             60255 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      3.505s
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -1464,11 +1464,56 @@ func (b *builtinSubDateDurationIntSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinYearWeekWithModeSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals YEARWEEK(date,mode).
```
Comment (b41sh) [inline expression/builtin_time_vec.go]: vecEvalInt

Comment (chenlx0) [reply expression/builtin_time_vec.go]: fixed

Comment (chenlx0) [reply expression/builtin_time_vec.go]: > vecEvalInt
Fixed.

Comment (chenlx0) [reply expression/builtin_time_vec.go]: Fixed.

**Comment by b41sh** [comment]
LGTM

**Pull request merged**